specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Patches.json

```diff
@@ -0,0 +1,380 @@
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Patches_Apply_0": {
+            "$ref": "./examples/Patches_Apply.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PatchResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PatchProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "description": "The resource representation of a patch.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Creating",
+        "Updating",
+        "Deleting"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchStatus": {
+      "type": "string",
+      "description": "The status of the patch once it has been provisioned",
+      "enum": [
+        "NotStarted",
+        "InProgress",
+        "Succeeded",
+        "Canceled",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "PatchStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchObject": {
+      "description": "Individual patch object.",
+      "type": "object",
+      "properties": {
+        "patchName": {
+          "type": "string",
+          "description": "The name of the patch."
+        },
+        "targetImage": {
+          "type": "string",
+          "description": "The name of the target image for the patch."
+        },
+        "newRunImage": {
+          "type": "string",
+          "description": "The name of new run image for the target image."
+        },
+        "oldRunImage": {
+          "type": "string",
+          "description": "The name of old run image for the target image."
+        },
+        "reason": {
+          "type": "string",
+          "description": "The reason for the patch."
+        },
```

Comment:
We could represent this as enum, but that limits us to just some constant reasons for patching. Say if there's an important security issue and requires immediate fix, the reason could be customizable through adding a reason message when rebuilding the detection image.

Reply:
perhaps we can split this into 2 fields (one enum for easy filtering and one more verbose message)? I agree with cormac that it would be nice to allow customers to filter by overall type of patch (for example whether its a security fix or not). But first off, do we even have that information available to us? An example of what would go into this field and what info we currently get from the patch detection may help us decide the format